File: Ahorn/entities/FeatherRefill.jl
module AnonhelperFeatherRefill

using ..Ahorn, Maple

@mapdef Entity "Anonhelper/FeatherRefill" FeatherRefill(x::Integer, y::Integer, oneUse::Bool=false)


const placements = Ahorn.PlacementDict(
    "Feather Refill (Anonhelper)" => Ahorn.EntityPlacement(
        FeatherRefill, 
	"point"
    )
)

sprite = "objects/featherRefill/idle00"

function getSprite(entity::FeatherRefill)

    return sprite
end

function Ahorn.selection(entity::FeatherRefill)
    oneUse = get(entity.data, "oneUse", false)
    x, y = Ahorn.position(entity)
    sprite = getSprite(entity)

    return Ahorn.getSpriteRectangle(sprite, x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::FeatherRefill, room::Maple.Room)
    sprite = getSprite(entity)
    Ahorn.drawSprite(ctx, sprite, 0, 0)
end

end

File: Loenn/entities/feather_refill.lua
local utils = require("utils")

local featherRefill = {}

featherRefill.name = "Anonhelper/FeatherRefill"
featherRefill.depth = -100
featherRefill.placements = {
    {
        name = "feather_refill",
        data = {
            oneUse = false,
        }
    }
}
featherRefill.texture = "objects/AnonHelper/featherRefill/idle00"

function featherRefill.selection(room, entity)
	local x, y = entity.x or 0, entity.y or 0
    return utils.rectangle(x - 4, y - 6, 8, 12)
end

return featherRefill